contributors/design-proposals/federated-deployment.md

```diff
@@ -0,0 +1,88 @@
+# Federated Deployment
+# Design Document
+
+Authors: [Marcin Wielgus](mailto:mwielgus@google.com), [Nikhil Jindal](mailto:nikhiljindal@google.com )
+
+## Introduction
+
+The purpose of this document is to provide a detailed design of how Kubernetes Deployments should be handled in Federation, with the special emphasis on rolling update support in deployment updates. 
+
+The internal design is based on [Federated ReplicaSet](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/federated-replicasets.md)
+
+## Background
+
+With Federation we are trying to provide Kubernetes objects and functionality spanning over multiple clusters in geographically distributed locations. One of the key objects in Kubernetes is Deployment. Deployment allows users to maintain a defined number of pod replicas and conveniently update them if needed. The update functionality is the main thing that differentiate them from replica sets. ReplicaSets update everything at once while Deployments allow a slow, rolling update and rollback in case the update is unsuccessful. 
+
+## Requirements
+
+Federated deployments must provide the similar functionality as regular deployments, in particular :
+  + [R1] Work well with kubectl rollout
+    + History - provide all of the revisions of federated deployment.
+    + Pause/Resume - provide an ability to pause and resume an ongoing update
+    + Status - provide rollout status information
+    + Undo - provide go back to the previous version
+    + Require no/almost-no fixes in kubectl
+  + [R2] Allow rolling update
+    + One cluster at a time (preferred)
+    + All clusters at a time (additional)
+  + [R3] Be similar to what is currently implemented in regular Kubernetes.
+  + [R4] Allow similar scheduling extra features as Federated ReplicaSet
+    + Specifying min/max/weight for each of the clusters
+    + Be able to rebalance replicas once a cluster is added, lost or if there is no capacity.
+  + [R5] Each of the underlying clusters should be able to work independently if the federation control plane is down and be as easily updatable as federation. This basically means that Federated Deployment should create fully-functional deployments in the clusters.
```

Comment:
I think it would be helpful to detail the use cases justifying this requirement.  Without it, Federated Deployment could be a lot simpler.